cosi.proto

```diff
@@ -0,0 +1,131 @@
+syntax = "proto3";
+package cosi.v1alpha1;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = "cosi";
+
+extend google.protobuf.EnumOptions {
+    // Indicates that this enum is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_enum = 1060;
+}
+
+extend google.protobuf.EnumValueOptions {
+    // Indicates that this enum value is OPTIONAL and part of an
+    // experimental API that may be deprecated and eventually removed
+    // between minor releases.
+    bool alpha_enum_value = 1060;
+}
+
+extend google.protobuf.FieldOptions {
+    // Indicates that a field MAY contain information that is sensitive
+    // and MUST be treated as such (e.g. not logged).
+    bool cosi_secret = 1059;
+
+    // Indicates that this field is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_field = 1060;
+}
+
+extend google.protobuf.MessageOptions {
+    // Indicates that this message is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_message = 1060;
+}
+
+extend google.protobuf.MethodOptions {
+    // Indicates that this method is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_method = 1060;
+}
+
+extend google.protobuf.ServiceOptions {
+    // Indicates that this service is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_service = 1060;
+}
+
+service Provisioner {
+    rpc ProvisionerGetInfo (ProvisionerGetInfoRequest) returns (ProvisionerGetInfoResponse) {}
+
+    rpc ProvisionerCreateBucket (ProvisionerCreateBucketRequest) returns (ProvisionerCreateBucketResponse) {}
+    rpc ProvisionerDeleteBucket (ProvisionerDeleteBucketRequest) returns (ProvisionerDeleteBucketResponse) {}
+
+    rpc ProvisionerGrantBucketAccess (ProvisionerGrantBucketAccessRequest) returns (ProvisionerGrantBucketAccessResponse);
+    rpc ProvisionerRevokeBucketAccess (ProvisionerRevokeBucketAccessRequest) returns (ProvisionerRevokeBucketAccessResponse);
+}
+
+message ProvisionerGetInfoRequest {
+    // Intentionally left blank
+}
+
+message ProvisionerGetInfoResponse {    
+    string name = 1;
+}
+
+message ProvisionerCreateBucketRequest {    
+    string bucket_name = 1;
```

Comment:
This is a general comment.  For every field, add a comment to explain what it means.  Also specify whether it is REQUIRED or OPTIONAL

Reply:
Fixed.